lib/ddtrace/contrib/rails/framework.rb

```diff
@@ -26,7 +26,8 @@ module Framework
           tracer: Datadog.tracer,
           debug: false,
           trace_agent_hostname: Datadog::Writer::HOSTNAME,
-          trace_agent_port: Datadog::Writer::PORT
+          trace_agent_port: Datadog::Writer::PORT,
+          env: ::Rails.env
```

Comment:
It looks like this change has yielded unexpected results, at least for me. With this change, some spans from the app—and _only some_ but not all, oddly—are getting `env:production` set.

I have a Rails application that we run in multiple named production-like environments, e.g. `staging` and `production`. Both systems' `Rails.env` is `production` per Rails conventions. I differentiate between them by setting the `env:` tag in the agent's `datadog.conf`.

At a minimum, I'd call this change backwards-incompatible: upgrading the gem results in data getting sent with an unexpected `env:` tag.

Was this change to handle deployments where a single Datadog agent on a host would be receiving and forwarding data from multiple applications (not just development/test/production) and so therefore the application needs a way to name its stuff (traces? spans? I'm new here.).

What do ya'll think of a more specific default as a combination of the Rails application name and the Rails environment?

```
> Rails.application.class.to_s.split("::").first.downcase + '-' + Rails.env
=> "myrailsapp-production"
```

I'm new to APMing with Datadog, so I fully admit I could be doing something wrong or could be doing something better.